### PR TITLE
Create GroupMembers table role

### PR DESCRIPTION
**Pull request by sam-codaio**
* Create GroupMembers concept in Packs SDK
* Requires 2 fields groupId and userId
* Sync Tables with this role will return a flattened list of all the members of a group
* If desired the developer can add a crawl strategy for this table that works on a 'groups' table and crawls members of each group individually. Alternatively this can just return every single group member from every single group in the system

### CODE REVIEW

File: schema.ts [hunk 1]
```diff
@@ -1225,10 +1228,22 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
...
+   * For cases where the object being synced represents a grou, the name of the property within
+   * {@link ObjectSchemaDefinition.properties} that identifies the id of the group in the service
+   * being synced from.
+   *
+   * This is required for sync tables with role {@link TableRole.GroupMembers}
+   *
+   * Must be a {@link ValueType.String} or {@link ValueType.Number} property
+   * @hidden
+   */
+  groupIdProperty?: PropertyIdentifier<K>;
```
Comment (sam-codaio) [inline schema.ts:1246]: Do we need a groupType if this system uses scim or google groups?

Comment (sam-codaio) [reply schema.ts:1246]: Actually we would not need that here, that would need to go on the permission.

File: schema.ts [hunk 2]
```diff
@@ -1225,10 +1228,22 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    * {@link ObjectSchemaDefinition.properties} that identifies the id of the user in the service
    * being synced from.
    *
+   * This is required for sync tables with role {@link TableRole.User} or {@link TableRole.GroupMembers}
    * Must be a {@link ValueType.String} or {@link ValueType.Number} property
    * @hidden
    */
   userIdProperty?: PropertyIdentifier<K>;
+  /**
+   * For cases where the object being synced represents a grou, the name of the property within
```
Comment (patrick-codaio) [inline schema.ts:1237]: you are likely to be eaten by a grou

**Review by patrick-codaio — APPROVED** [review]

File: schema.ts [hunk 3]
```diff
@@ -1225,10 +1228,22 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
...
+   * {@link ObjectSchemaDefinition.properties} that identifies the id of the group in the service
+   * being synced from.
+   *
+   * This is required for sync tables with role {@link TableRole.GroupMembers}
+   *
+   * Must be a {@link ValueType.String} or {@link ValueType.Number} property
+   * @hidden
```
[the remark this answers is not in the document]
Comment (sam-codaio) [reply schema.ts:1244]: Ahh forgot a TODO here. Will add

[the remark this answers is not in the document]
Comment (sam-codaio) [reply schema.ts:1244]:
```suggestion
   * TODO(sam): Unhide this
   * @hidden
```

**Pull request merged**